### PR TITLE
Update `CODEOWNERS` paths: fix invalid paths and deprioritize `/eng/` and Engineering System Team paths

### PR DESCRIPTION
**Pull request by konrad-jamrozik**
As part of ongoing work of enabling wildcard support for `CODEOWNERS`:
- https://github.com/Azure/azure-sdk-tools/issues/2770
- https://github.com/Azure/azure-sdk-tools/pull/5088

and enabling stricter validation:
- https://github.com/Azure/azure-sdk-tools/issues/4859

this PR:
- fixes invalid paths, to match rules explained [here](https://github.com/Azure/azure-sdk/blob/main/docs/policies/opensource.md#codeowners);
- deprioritizes ownership rules for `/eng/` dir and Azure SDK Engineering System team , to avoid all build failure notifications being routed to it once we enable the new regex-based, wildcard-supporting `CODEOWNERS` matcher, per: https://github.com/Azure/azure-sdk-tools/pull/5088#issuecomment-1397330147

As a consequence of this PR possibly the automatically assigned PR reviewers will change for some paths. The full list of of such changes can be determined by running relevant test provided by this PR:
- https://github.com/Azure/azure-sdk-tools/pull/5266

You can also reach out to me to get that information.

Once this PR is merged, I will enable the new `CODEOWNERS` matcher, similar to how it was done for `net` repo by these two PRs:
- https://github.com/Azure/azure-sdk-tools/pull/5241
- https://github.com/Azure/azure-sdk-tools/pull/5240

Doing so will also change to whom build failure notification emails are sent.

Related PRs:
- Similar PR fixing invalid paths, but for `net` repo: https://github.com/Azure/azure-sdk-for-net/pull/33584
- Similar PR deprioritizing the Azure SDK EngSys team ownership, but for `python` repo: https://github.com/Azure/azure-sdk-for-python/pull/28534

## Build failure notification recipients changes

For a build definition whose source is `PATH`, now its build failure notification recipients will be `RIGHT OWNERS`, while previously they where `LEFT OWNERS`:

image

### CODE REVIEW

**Review by benbp — APPROVED** [review]

File: .github/CODEOWNERS [hunk 1]
```diff
@@ -68,141 +99,141 @@
...
-/sdk/containerregistry/                             @lmolkova @alzimmermsft
+/sdk/containerregistry/                              @lmolkova @alzimmermsft
 
 # PRLabel: %Azure.Core
-/sdk/core/                                          @alzimmermsft @srnagar @anuchandy @lmolkova @vcolin7 @mssfang @jonathangiles @billwert
+/sdk/core/                                           @alzimmermsft @srnagar @anuchandy @lmolkova @vcolin7 @mssfang @jonathangiles @billwert
 
 # PRLabel: %Azure.Core.AMQP
-/sdk/core/azure-core-amqp/                          @conniey @anuchandy @ki1729 @srnagar @jonathangiles @haolingdong-msft @liukun-msft @weidongxu-microsoft @ZejiaJiang
+/sdk/core/azure-core-amqp/                           @conniey @anuchandy @ki1729 @srnagar @jonathangiles @haolingdong-msft @liukun-msft @weidongxu-microsoft @ZejiaJiang
```
Comment (konrad-jamrozik) [inline .github/CODEOWNERS:114]: Previously `/sdk/core/azure-core-amqp-experimental/` matched against this path, but now it matches against `/sdk/core/`.

File: .github/CODEOWNERS [hunk 2]
```diff
@@ -68,141 +99,141 @@
...
 
 # PRLabel: %Azure.Core
-/sdk/core/azure-core-tracing-opentelemetry/         @samvaity @alzimmermsft @trask @lmolkova
+/sdk/core/azure-core-tracing-opentelemetry/          @samvaity @alzimmermsft @trask @lmolkova
```
Comment (konrad-jamrozik) [inline .github/CODEOWNERS:117]: Previously `/sdk/core/azure-core-tracing-opentelemetry-samples/` matched against this path, but now it matches against `/sdk/core/`.

File: .github/CODEOWNERS [hunk 3]
```diff
@@ -68,141 +99,141 @@
...
-/sdk/monitor/azure-monitor-opentelemetry-exporter   @trask @ramthi @heyams @jeanbisutti
+/sdk/monitor/azure-monitor-opentelemetry-exporter/   @trask @ramthi @heyams @jeanbisutti
 
 # PRLabel:  %Monitor
-/sdk/monitor/azure-monitor-query                    @srnagar @lmolkova @ki1729
+/sdk/monitor/azure-monitor-query/                    @srnagar @lmolkova @ki1729
 
 # PRLabel:  %Monitor
-/sdk/monitor/azure-monitor-ingestion                @srnagar @lmolkova @ki1729
+/sdk/monitor/azure-monitor-ingestion/                @srnagar @lmolkova @ki1729
```
Comment (konrad-jamrozik) [inline .github/CODEOWNERS:181]: Do we also want a rule for `/sdk/monitor/azure-monitor-ingestion-perf/` ? Similar with `/sdk/monitor/azure-monitor-query-perf/`.

File: .github/CODEOWNERS [hunk 4]
```diff
@@ -1,47 +1,78 @@
...
+# As of 1/30/2023 these paths have no owners:
+
+# /
+# /.devcontainer/
+# /.vscode/
+# /common/perf-test-score/
+# /doc/
+# /samples/
+
+###########
```
Comment (weshaggard) [inline .github/CODEOWNERS:16]: Why did this get moved up top? Also why haven't we removed `**/ci.yml` and `**/tests.yml` paths?

File: .github/CODEOWNERS [hunk 5]
```diff
@@ -68,141 +99,141 @@
...
-/sdk/monitor/azure-monitor-opentelemetry-exporter   @trask @ramthi @heyams @jeanbisutti
+/sdk/monitor/azure-monitor-opentelemetry-exporter/   @trask @ramthi @heyams @jeanbisutti
 
 # PRLabel:  %Monitor
-/sdk/monitor/azure-monitor-query                    @srnagar @lmolkova @ki1729
+/sdk/monitor/azure-monitor-query*/                  @srnagar @lmolkova @ki1729
 
 # PRLabel:  %Monitor
-/sdk/monitor/azure-monitor-ingestion                @srnagar @lmolkova @ki1729
+/sdk/monitor/azure-monitor-ingestion*/              @srnagar @lmolkova @ki1729
```
Comment (konrad-jamrozik) [inline .github/CODEOWNERS:181]: @srnagar @lmolkova @ki1729 I want to bring your attention to these two changes. This means you are also taking ownership of contents of 
`/sdk/monitor/azure-monitor-query-perf/`
and
`/sdk/monitor/azure-monitor-ingestion-perf/`

Previously these directories matched to `/sdk/` owned by `joshfree,srnagar,anuchandy,conniey,lmolkova,jonathangiles`

If this is undesired behavior, please do let me know, I'll rever the change.

CC @weshaggard @jsquire

File: .github/CODEOWNERS (hunk 4 above)
Comment (konrad-jamrozik) [reply .github/CODEOWNERS:16]: Fixed by:
- https://github.com/Azure/azure-sdk-for-java/pull/33309

**Pull request merged**